packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -170,7 +170,16 @@ export const getDecoratedVariationAttributes = (product, opts = {}) => {
  */
 export const findLowestPrice = (product) => {
     if (!product) return
-    const array = product.variants ?? [product]
+
+    // Look at all of the variants, only if it's a master product.
+    // i.e. when a shopper has narrowed down to a variant, do not look into other variants
+    const isMaster = product.productType?.master || product.type?.master
+    if (isMaster && !product.variants) {
+        console.warn(
+            'Expecting `product.variants` to exist. For more accuracy, please tweak your API request to include allVariationProperties=true'
```

Comment:
Do we really need to specify the API query parameter here `allVariationProperties=true`. If the API spec changes in future then our message will become obsolete. May be we can keep the message generic.. " Check the API documentation to include variant details"